### ISSUE TITLE
weird --log.level=DEBUG behavior (upd: reason is in networks)

### ISSUE THREAD

**Issue by vanodevium** [traefik/traefik#10100]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I want to setup traefik proxying to my applications from another folders and own compose files.

All settings I take from official traefik and docker sites.

### What did you see instead?

When i make ping from traefik service, I see response from application.
But from browsers I see only 504.

The weird thing is:
**when I enable `--log.level=DEBUG`, traefik works correctly from container and browsers.**

### What version of Traefik are you using?

Version:      2.10.4
Codename:     saintmarcelin
Go version:   go1.20.6
Built:        2023-07-24T16:29:02Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

Traefik compose file:
```yaml
version: "3.7"

services:
  traefik:
    image: "traefik:latest"
    container_name: "traefik"
    command:
      - "--log.level=DEBUG"
      #- "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.https.address=:443"
    ports:
      - "80:80"
      #- "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    networks:
      - traefik

networks:
  traefik:
    name: "test-network"
    driver: bridge
```

Application compose file:
```
version: "3.7"

services:
  application:
    container_name: application
    build: .
    restart: unless-stopped
    env_file: .env
    networks:
      - app
      - traefik
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.test.rule=Host(`test.localhost`)"
      - "traefik.http.routers.test.entrypoints=web"

networks:
  app:
    name: "app"
    driver: bridge
  traefik:
    name: "test-network"
    external: true

```

### If applicable, please paste the log output in DEBUG level

```sh
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Configuration loaded from flags."
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Traefik version 2.10.4 built on 2023-07-24T16:29:02Z"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"https\":{\"address\":\":443\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{},\"http2\":{\"maxConcurrentStreams\":250},\"udp\":{\"timeout\":\"3s\"}},\"web\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{},\"http2\":{\"maxConcurrentStreams\":250},\"udp\":{\"timeout\":\"3s\"}}},\"providers\":{\"providersThrottleDuration\":\"2s\",\"docker\":{\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"swarmModeRefreshSeconds\":\"15s\"}},\"log\":{\"level\":\"DEBUG\",\"format\":\"common\"}}"
traefik  | time="2023-08-29T15:23:56Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://doc.traefik.io/traefik/contributing/data-collection/\n"
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Starting provider aggregator aggregator.ProviderAggregator"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Starting TCP Server" entryPointName=https
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Starting TCP Server" entryPointName=web
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Starting provider *traefik.Provider"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="*traefik.Provider provider configuration: {}"
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Starting provider *docker.Provider"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="*docker.Provider provider configuration: {\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"swarmModeRefreshSeconds\":\"15s\"}"
traefik  | time="2023-08-29T15:23:56Z" level=info msg="Starting provider *acme.ChallengeTLSALPN"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="*acme.ChallengeTLSALPN provider configuration: {}"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Configuration received: {\"http\":{\"services\":{\"noop\":{}},\"serversTransports\":{\"default\":{\"maxIdleConnsPerHost\":200}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=internal
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Provider connection established with docker 24.0.2 (API 1.43)" providerName=docker
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Filtering disabled container" providerName=docker container=traefik-outbid-proxy-aee9445a7877692433269c344faadab50c7b4423a482eb37403b611b6fb409cb
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Configuration received: {\"http\":{\"routers\":{\"test\":{\"entryPoints\":[\"web\"],\"service\":\"application-test\",\"rule\":\"Host(`test.localhost`)\"}},\"services\":{\"application-test\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.112.2:3000\"}],\"passHostHeader\":true}}}},\"tcp\":{},\"udp\":{}}" providerName=docker
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="No default certificate, fallback to the internal generated certificate" tlsStoreName=default
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="No default certificate, fallback to the internal generated certificate" tlsStoreName=default
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Creating middleware" routerName=test@docker serviceName=application-test entryPointName=web middlewareName=pipelining middlewareType=Pipelining
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Creating load-balancer" entryPointName=web routerName=test@docker serviceName=application-test
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Creating server 0 http://192.168.112.2:3000" routerName=test@docker serviceName=application-test serverName=0 entryPointName=web
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="child http://192.168.112.2:3000 now UP"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Propagating new UP status"
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Added outgoing tracing middleware application-test" entryPointName=web routerName=test@docker middlewareName=tracing middlewareType=TracingForwarder
traefik  | time="2023-08-29T15:23:56Z" level=debug msg="Creating middleware" middlewareType=Recovery entryPointName=web middlewareName=traefik-internal-recovery
```

**Comment by vanodevium** [comment]
**More observations:**

- when I start traefik, then application - it's not working, but after restart traefik - OK
- when I start application only, then traefik - it is working OK

--- 

Maybe it isn't problem with _--log.level_, but with _networks_

**Comment by vanodevium** [comment]
As it turned out, the problem is really in the networks.

When I restarted the container with traefik, I changed the log.level, but it was not the logs that helped, but the restart itself.

The main reason was that the application container had two networks and the traefik did not quite understand which IP to contact the application.

--- 

To fix this, simply add to the application container label:
```yaml
- "traefik.docker.network=traefik"
```
> where traefik is name of network where traefik run as container.
